connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -607,8 +608,18 @@ private WorkerTask buildWorkerTask(ClusterConfigState configState,
                                             Class<? extends Connector> connectorClass,
                                             ConnectorClientConfigOverridePolicy connectorClientConfigOverridePolicy) {
         Map<String, Object> adminProps = new HashMap<>();
-        adminProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, Utils.join(config.getList(WorkerConfig.BOOTSTRAP_SERVERS_CONFIG), ","));
-        // User-specified overrides
+        // Use the top-level worker configs to retain backwards compatibility with older releases which
+        // did not require a prefix for connector admin client configs in the worker configuration file
+        // Only include configs that do not begin with "admin." since those will be added next, but with
+        // the prefix stripped
+        Map<String, Object> nonPrefixedWorkerConfigs = config.originals().entrySet().stream()
+            .filter(e -> !e.getKey().startsWith("admin."))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        adminProps.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG,
+            Utils.join(config.getList(WorkerConfig.BOOTSTRAP_SERVERS_CONFIG), ","));
+        adminProps.putAll(nonPrefixedWorkerConfigs);
+
+        // Admin client-specific overrides in the worker config
```

Comment:
One concern I have with this approach (which is technically the correct direction) is that it puts *all worker properties* into the admin properties. Remember, all Kafka clients log warnings for all configs unnecessarily provided to them. So, should we exclude as many configuration properties as we can that we know will not be used for the worker's admin properties, such as `producer.*` and `consumer.*` (which are for the connector's clients)? Should we also eliminate any of the known worker configs that are not used in the admin client?

Reply:
I'm not completely opposed to trying to remove unnecessary warn-level log messages but I'm wondering if this is the right place and method for going about that. It looks like https://issues.apache.org/jira/browse/KAFKA-7509 still has a fix open with https://github.com/apache/kafka/pull/5876 that touches on more than just admin client configs for the DLQ. Would it be better to try to address that issue in two stages with a partial measure in this PR and a more complete approach in https://github.com/apache/kafka/pull/5876 (or at least, would it be beneficial enough to justify the split), or should we keep these two issues separate and implement fixes for them separately?

FWIW, I _hate_ the unnecessary warning logs that get generated every time a worker or connector starts up. If all that's needed for https://github.com/apache/kafka/pull/5876 to move forward is a rebase and a pair of eyes, I'm happy to help there. Just wondering how much value would be added by removing some but not all or even most of these warning logs.

Reply:
I agree with @C0urante that general filtering is a subject of another PR. 
But I also agree with @rhauch that we should exclude any properties that we know that they'll never be useful with a certain prefix. And indeed such properties are the ones starting with `producer.` and `consumer.` here. So an ORed `startsWith` would make sense to me.

Reply:
@kkonstantine, thanks for the correction -- the admin properties for the sink worker task (and specifically the DLQ reporter for the sink task) should use a combination of the `producer.*` and `consumer.*` properties that are connector-specific.